client_it_test.go

```
@@ -92,6 +95,163 @@ func TestClientLifecycleEvents(t *testing.T) {
+			t.Fatal(err)
+		}
+	}(ctx, client)
+	subscriptionID, err := client.AddMembershipListener(func(event cluster.MembershipStateChanged) {
+		switch event.State {
+		case cluster.MembershipStateAdded:
+			t.Logf("MembershipStateAdded")
+			wgAdded.Done()
+		case cluster.MembershipStateRemoved:
+			t.Logf("MembershipStateRemoved")
+			wgRemoved.Done()
+		}
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	assert.NotEqual(t, types.UUID{}, subscriptionID, "subscription UUID should not be empty")
+	// member which is not connected by the client initially.
+	uuid := cls.MemberUUIDs[1]
+	//uuid := cls.MemberUUIDs[0]
+	ok, err := cls.RC.TerminateMember(ctx, cls.ClusterID, uuid)
+	it.Must(err)
+	assert.True(t, ok, "rc cannot terminate member")
+	wgRemoved.Wait()
+	_, err = cls.RC.StartMember(ctx, cls.ClusterID)
+	it.Must(err)
+	wgAdded.Wait()
+}
+
+func TestClient_RemoveMembershipListener(t *testing.T) {
+	var added int32 = 1
+	ctx := context.Background()
+	cls := it.StartNewClusterWithOptions("client-subscribed-add-membership-listener", 15701, 2)
+	cfg := cls.DefaultConfig()
+	client, err := hz.StartNewClientWithConfig(ctx, cfg)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer func(ctx context.Context, client *hz.Client) {
+		err = client.Shutdown(ctx)
+		if err != nil {
+			t.Fatal(err)
+		}
+	}(ctx, client)
+	subscriptionID, err := client.AddMembershipListener(func(event cluster.MembershipStateChanged) {
+		switch event.State {
+		case cluster.MembershipStateRemoved:
+			t.Logf("MembershipStateRemoved")
+			atomic.SwapInt32(&added, 0)
+		}
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	assert.NotEqual(t, types.UUID{}, subscriptionID, "subscription UUID should not be empty")
+	err = client.RemoveMembershipListener(subscriptionID)
+	if err != nil {
+		t.Fatal(err)
+	}
+	ok, err := cls.RC.TerminateMember(ctx, cls.ClusterID, cls.MemberUUIDs[1])
+	it.Must(err)
+	assert.True(t, ok, "rc cannot terminate member")
+	go func() {
+		time.Sleep(time.Minute * 1)
+		_ = client.Shutdown(ctx)
```

Could you add a comment about the error is ignored.

---

Fixed: [88b6f60](https://github.com/hazelcast/hazelcast-go-client/pull/809/commits/88b6f60a915c3d0237ae5f1fb97bdbda52cedfbd)